sdk/cognitiveservices/azure-cognitiveservices-language-textanalytics/azure/cognitiveservices/language/textanalytics/_models.py

```diff
@@ -0,0 +1,534 @@
+     field will contain information about the document payload.
+    :type statistics:
+     ~azure.cognitiveservices.language.textanalytics.models.DocumentStatistics
+    :param bool is_error: Boolean check for error item when iterating over list of
+     results. Always False for an instance of a DocumentKeyPhrases.
+    """
+
+    def __init__(self, **kwargs):
+        self.id = kwargs.get("id", None)
+        self.key_phrases = kwargs.get("key_phrases", None)
+        self.statistics = kwargs.get("statistics", None)
+        self.is_error = False
+
+
+class DocumentLinkedEntities(DictMixin):
+    """DocumentLinkedEntities.
+
+    :param id: Unique, non-empty document identifier.
+    :type id: str
+    :param entities: Recognized well-known entities in the document.
+    :type entities:
+     list[~azure.cognitiveservices.language.textanalytics.models.LinkedEntity]
+    :param statistics: If show_stats=true was specified in the request this
+     field will contain information about the document payload.
+    :type statistics:
+     ~azure.cognitiveservices.language.textanalytics.models.DocumentStatistics
+    :param bool is_error: Boolean check for error item when iterating over list of
+     results. Always False for an instance of a DocumentLinkedEntities.
+    """
+
+    def __init__(self, **kwargs):
+        self.id = kwargs.get("id", None)
+        self.entities = kwargs.get("entities", None)
+        self.statistics = kwargs.get("statistics", None)
+        self.is_error = False
+
+
+class DocumentSentiment(DictMixin):
+    """DocumentSentiment.
+
+    :param id: Unique, non-empty document identifier.
+    :type id: str
+    :param sentiment: Predicted sentiment for document (Negative,
+     Neutral, Positive, or Mixed). Possible values include: 'positive',
+     'neutral', 'negative', 'mixed'
+    :type sentiment: str
+    :param statistics: If show_stats=true was specified in the request this
+     field will contain information about the document payload.
+    :type statistics:
+     ~azure.cognitiveservices.language.textanalytics.models.DocumentStatistics
+    :param document_scores: Document level sentiment confidence
+     scores between 0 and 1 for each sentiment class.
+    :type document_scores: dict
+    :param sentences: Sentence level sentiment analysis.
+    :type sentences:
+     list[~azure.cognitiveservices.language.textanalytics.models.SentenceSentiment]
+    :param bool is_error: Boolean check for error item when iterating over list of
+     results. Always False for an instance of a DocumentSentiment.
+    """
+
+    def __init__(self, **kwargs):
+        self.id = kwargs.get("id", None)
+        self.sentiment = kwargs.get("sentiment", None)
+        self.statistics = kwargs.get("statistics", None)
+        self.document_scores = kwargs.get("document_scores", None)
+        self.sentences = kwargs.get("sentences", None)
+        self.is_error = False
```

Comment:
Since the batch methods return a heterogeneous list of result objects and error objects, it just makes it easier to filter through the list like this:
`docs = [doc for doc in result if not doc.is_error]`

For `is_error` a result object will always be False and an error object will be True.